ouroboros-consensus/src/ouroboros-consensus/Ouroboros/Consensus/Ledger/Tables.hs

```diff
@@ -295,13 +295,15 @@ valuesMKDecoder ::
 valuesMKDecoder = do
     _ <- CBOR.decodeListLenOf 1
     mapLen <- CBOR.decodeMapLen
-    LedgerTables <$> go mapLen
+    LedgerTables . ValuesMK <$> go mapLen Map.empty
  where
   go :: Int
-     -> CBOR.Decoder s (ValuesMK (TxIn l) (TxOut l))
-  go len =
-    ValuesMK . Map.fromList
-      <$> replicateM len (unpackError @(TxIn l, TxOut l) @ByteString <$> fromCBOR)
+     -> Map.Map (TxIn l) (TxOut l)
+     -> CBOR.Decoder s (Map.Map (TxIn l) (TxOut l))
+  go 0 m = pure m
+  go !len m = do
```

Comment:
I think the only place you need bangs are `len` and `m`, yeah? Since you're using the `Data.Map.Strict` interface. (Which would plug the leak in `m` here, also.)